expression/schema.go

```diff
@@ -90,8 +90,22 @@ func (s *Schema) Clone() *Schema {
 
 // ExprFromSchema checks if all columns of this expression are from the same schema.
 func ExprFromSchema(expr Expression, schema *Schema) bool {
-	cols := ExtractColumns(expr)
-	return len(schema.ColumnsIndices(cols)) > 0
+	switch v := expr.(type) {
+	case *Column:
+		return schema.Contains(v)
+	case *CorrelatedColumn:
```

Comment:
a correlated column is treated as a constant, not a column.
In `t.a=xx.a` if `xx.a` is a correlated column, this is a one side condition rather than other compound condition.